test_util/test_upgrade_vpc.py

```diff
@@ -159,125 +132,279 @@ def wait_for_dns(hostname):
+            # trying to resolve is available.
+            self.wait_for_dns(dcos_api, dns_app['env']['RESOLVE_NAME'])
+            dcos_api.marathon.deploy_app(dns_app, check_health=False)
+            dcos_api.marathon.ensure_deployments_complete()
+
+            test_apps = [healthcheck_app, dns_app]
+            self.test_app_ids = [app['id'] for app in test_apps]
+
+            self.tasks_start = {app_id: sorted(self.app_task_ids(dcos_api, app_id)) for app_id in self.test_app_ids}
+            self.log.debug('Test app tasks at start:\n' + pprint.pformat(self.tasks_start))
+
+            for app in test_apps:
+                assert app['instances'] == len(self.tasks_start[app['id']])
+
+    def verify_apps_state(self, dcos_api: DcosApiSession, dns_app: dict):
+        with logger.scope("verify apps state"):
+
+            # nested methods here so we can "close" over external state
+
+            def marathon_app_tasks_survive_upgrade():
+                # Verify that the tasks we started are still running.
+                tasks_end = {app_id: sorted(self.app_task_ids(dcos_api, app_id)) for app_id in self.test_app_ids}
+                self.log.debug('Test app tasks at end:\n' + pprint.pformat(tasks_end))
+                if not self.tasks_start == tasks_end:
+                    self.teamcity_msg.testFailed(
+                        "test_upgrade_vpc.marathon_app_tasks_survive_upgrade",
+                        details="expected: {}\nactual:   {}".format(self.tasks_start, tasks_end))
+
+            def test_app_dns_survive_upgrade():
+                # Verify DNS didn't fail.
+                marathon_framework_id = dcos_api.marathon.get('/v2/info').json()['frameworkId']
+                dns_app_task = dcos_api.marathon.get('/v2/apps' + dns_app['id'] + '/tasks').json()['tasks'][0]
+                dns_log = self.parse_dns_log(dcos_api.mesos_sandbox_file(
+                    dns_app_task['slaveId'],
+                    marathon_framework_id,
+                    dns_app_task['id'],
+                    dns_app['env']['DNS_LOG_FILENAME'],
+                ))
+                dns_failure_times = [entry[0] for entry in dns_log if entry[1] != 'SUCCESS']
+                if len(dns_failure_times) > 0:
+                    message = 'Failed to resolve Marathon app hostname {} at least once.'.format(
+                        dns_app['env']['RESOLVE_NAME'])
+                    err_msg = message + ' Hostname failed to resolve at these times:\n' + '\n'.join(dns_failure_times)
+                    self.log.debug(err_msg)
+                    self.teamcity_msg.testFailed("test_upgrade_vpc.test_app_dns_survive_upgrade", details=err_msg)
+                    # Skip raising an exception on DNS failure so that other tests can run.
+                    # DNS failure is being tracked at
+                    # https://mesosphere.atlassian.net/browse/DCOS-13426.
+                    # raise Exception(message)  # noqa
```

Comment:
Since we're now failing the test in TeamCity, we can get rid of the above comments.